docs/probes.md

```diff
@@ -0,0 +1,130 @@
+# Readiness and LivenessProbes
+
+The VMI spec allows setting `livenessProbe` and `readinessProbe` which translate to the same field on the resulting pod running the VM.
+
+## Exec Probes
+
+One option to probe the VM is by running commands on it and determine the ready/live state based on it's success.
+
```

Comment:
worth noting in this documentation that this probe behavior requires not only the guest agent to be installed, but also for the guest-exec api to be enabled within the guest.

For security reasons, this guest agent command is often disabled by default. For example, rhel8 disables it by default

Reply:
Right, I'm still investigating if we can provide images (like the new containerDisks) with the agent installed and enabled by default or at least get it there for some dists frequently used (like Fedora Cloud).

Reply:
i don't know how feasible that is.  the agent maintainers are pushing out builds using rpms for respective distros like fedora, rhel, suse, etc... If we come up wiht our own delivery mechanism, we have to continually build and push out new containers with the agent in it.  then we'd have to match those builds with OS versions. for example, an agent built for the new fedora 34 won't work wiht fedora 32 due to glibc requirements. 

i think the simpliest thing would be to document this requires qemu-exec to be enabled and provide an example of how to do that.